File: backend/src/main/java/com/kiwi/features/metrics/MetricsRepository.java
package com.kiwi.features.metrics;

import com.kiwi.features.users.UsersPersistence;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.time.LocalDate;
import java.util.Optional;

@Repository
public interface MetricsRepository extends JpaRepository<MetricsPersistence, Long> {
    Optional<MetricsPersistence> findByUserAndDate(UsersPersistence usersPersistence, LocalDate date);
}

File: mobile/app/src/main/java/com/bellako/kiwi/features/metrics/model/MetricsRepository.kt
package com.bellako.kiwi.features.metrics.model

import android.os.Build
import androidx.annotation.RequiresApi
import com.bellako.kiwi.common.utils.DateUtils.dateToString
import com.bellako.kiwi.features.metrics.data.MetricsDTO
import java.time.LocalDate

class MetricsRepository(
    private val api: IMetricsAPI,
) {
    suspend fun createMetrics(dto: MetricsDTO): Result<MetricsDTO> =
        runCatching {
            api.createMetrics(dto)
        }

    suspend fun updateMetrics(dto: MetricsDTO): Result<MetricsDTO> =
        runCatching {
            api.updateMetrics(dto)
        }

    @RequiresApi(Build.VERSION_CODES.O)
    suspend fun getMetricsByDate(date: LocalDate): Result<MetricsDTO?> =
        runCatching {
            api.getMetricsByDate(dateToString(date))
        }
}
